loopy/schedule/__init__.py

```diff
@@ -216,40 +215,39 @@ def find_loop_nest_around_map(kernel):
     """Returns a dictionary mapping inames to other inames that are
     always nested around them.
     """
-    result = {}
+    from functools import reduce
+    from collections import defaultdict
+    from loopy.schedule.tools import get_loop_nest_tree
 
-    all_inames = kernel.all_inames()
+    tree = get_loop_nest_tree(kernel)
 
-    iname_to_insns = kernel.iname_to_insns()
+    loop_nest_around_map = defaultdict(frozenset)
```

Comment:
In my view, the `loop_nest_around_map` should go. The loop nest tree is a strictly more informative data structure, and it is not of size O(n^2). I think we should get rid of it. I looked through the uses, and it shouldn't be hard to replace, and it'll fix another bottleneck in linearization.

Reply:
In the new version, I didn't touch the old implementation. So, for such cases $O(n^2)$ stays.

Reply:
I can handle it as a separate PR.